internal/pack/pack.go

```diff
@@ -129,6 +129,13 @@ func (p *Packer) Size() uint {
 	return p.bytes
 }
 
+// HeaderFull returns true if the pack header is at maximum capacity.
+func (p *Packer) HeaderFull() bool {
+	p.m.Lock()
+	defer p.m.Unlock()
+	return EntrySize+crypto.Extension+(uint(headerLengthSize)*uint(len(p.blobs)-1)) <= uint(maxHeaderSize)
```

Comment:
I think the condition should be ` uint(headerLengthSize) + crypto.Extension + (len(p.blobs)+1)*uint(EntrySize) > uint(maxHeaderSize)`

`headerLengthSize` is only the length field. But the pack header also includes the blob type and its ID. And the result of the check, currently seems to be inverted.

Reply:
Agreed, re-evaluated, I think it'll be good with your check suggestion.